ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,94 @@
+  * This configuration allows any customization of the generated module.xml, making it very flexible to cover broader cases, for example, it is possible to declare module dependencies, mark them as exported, declare the main class and so on.
+  * This command is not appropriate for use in a managed domain or when connecting to the management CLI remotely. Modules should be added and removed manually in a production environment.
+
+This RFE is opened to have an alternative new operation to address simpler cases, where it is not required any additional configuration beyond on the default module.xml configurations, and supported in production environments and domain mode.
+
+The simplest way to accomplished this is to use the EE subsystem to configure a path to the shared folder. An example of this configuration could be:
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-add(name=global-folders,value={name=my-common-libs, path=/Users/me/libs})
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee:list-add(name=global-folders,value={name=my-common-libs, path=/Users/me/libs})
+
+The name and path attributes will be mandatory.
+Internally, this operation will create a module with the name `my-common-libs` that will have as resource-roots the folder `/Users/me/libs/` and all the jars files included in this folder. Subdirectories will not be scanned, but if it is required, users can create a new global-folder using the subdirectory path.
+
+The module created from the shared folder will be loaded as soon as the first application is deployed in the server. That means, if the server is started/restarted and there are no applications deployed, then the shared folder is neither scanned nor the module loaded.
+
+
+Removal of a global-folder could be done using the following operation:
+
+  [standalone@localhost:9990 /] /subsystem=ee:list-remove(name=global-folders,value={name=my-common-libs})
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee:list-remove(name=global-folders,value={name=my-common-libs})
+
+This RFE will not cover the availability of hot JBoss modules redeployments, that means, modifications of existing jar files as well as adding a new jar or removal an existing one will require a complete server restart, however, changes in plain resources as property files will be available immediately to the applications.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.jboss.org/browse/EAP7-1124[EAP7-1124]
+  * https://issues.jboss.org/browse/WFLY-1160[WFLY-1160]
+* https://issues.jboss.org/browse/WFCORE-4375[WFCORE-4375]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/MODULES-99[MODULES-99]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:vmarek@redhat.com[Vratislav Marek]
+
+
+=== Testing By
+[ ] Engineering
+
+[X] QE
```

Comment:
Basic unit tests for the functionality must be prepared by Engineering.
The more complex integration testing will be covered by QE.
In case of need to be the complete testing cover only by QE, you must give a reason why.